spark/dpl-interpreter/src/main/java/com/teragrep/pth_07/DPLInterpreter.java

```diff
@@ -292,4 +292,26 @@ public Scheduler getScheduler() {
     public List<InterpreterCompletion> completion(String buf, int cursor, InterpreterContext interpreterContext) {
         return null;
     }
+
+    @Override
+    public String getDataset(String noteId, String paragraphId, int start, int length, String searchString, int draw) throws InterpreterException{
+        if(notebookParagraphUserInterfaceManager == null){
+            throw new InterpreterException("DPLInterpreter's notebookParagraphUserInterfaceManager map is not instantiated!");
+        }
+        if(!notebookParagraphUserInterfaceManager.containsKey(noteId)){
+            throw new InterpreterException("DPLInterpreter does not have a UserInterfaceManager for note id "+noteId);
+        }
+        if(!notebookParagraphUserInterfaceManager.get(noteId).containsKey(paragraphId)){
+            throw new InterpreterException("DPLInterpreter does not have a UserInterfaceManager for paragraph id "+paragraphId+" within note id "+noteId);
+        }
+        if(notebookParagraphUserInterfaceManager.get(noteId).get(paragraphId).getDtTableDatasetNg() == null){
```

Comment:
I think there is a possible TOCTOU issue with getting and re-getting them in multithreaded context as they could be potentially changed between check and get. Maybe store results of gets in local variables?